### ISSUE TITLE
Remove `format` option from config

### ISSUE THREAD

**Issue by tisAliG** [sugarcoat/sugarcoat#32]
Instead of having the `format` option in the config, it should only be a flag in the CLI portion of SC. 

Examples: 
 - if you run in CLI: sugarcoat config.js, then you get the default functionality where we create an HTML file with the output to the path that is given in the `dest`.
 - if you run in CLI: sugarcoat --json config.js, then `dest` becomes inessential, and if a user still includes a `dest` we will ignore it, and output the JSON directly into the CLI. 

This way when you run SC as a module, you don't have to worry about setting the output, and the user shouldn't need to output anything but HTML.

**Comment by tisAliG** [comment]
@vbalcom @ryanfitzer, 
When the user runs `sugarcoat --json config.js`, should we assume that the user won't be giving us any templates, partials or assets (from within the settings.template object)? If the user just wants JSON, don't they just want to use sugarcoat for the parsing of their files? Or is that too large of an assumption?

**Comment by ryanfitzer** [comment]
> Or is that too large of an assumption?

We either ignore those parts of the config, or we don't.

Or, we remove the CLI.

**Comment by tisAliG** [comment]
So after conversing, we have come the conclusion (for now) that `dest` is required (as it currently is) and it's options are `'none'` or `'path/to/dest/'`. That way if the user wants to change the expanded config (the return from the promise) in any way (ex. to JSON) they must do it themselves in a `.then` statement after sugarcoat. We will be removing CLI at the moment, as @ryanfitzer has stated above.

**Comment by tisAliG** [comment]
Waiting to complete this as a lot of @vbalcom's updates in PR #36 will create merge conflicts. Once that is merged into master I will get back on this and fix the conflicts and then move forward.
